front/src/modules/rollingStock/helpers/utils.ts

```diff
@@ -193,5 +198,26 @@ export const orderSelectorList = (list: (string | null)[]) => {
     : list.sort();
 };
 
+export const sortSelectedModeCurves = (curvesList: ConditionalEffortCurve[]) => {
+  const [allConds, onlyPR, onlyEP, nulls] = curvesList.reduce(
+    (sortedCurves, curve) => {
+      const { cond } = curve;
+      let arrIndex = 0;
+      if (!cond?.electrical_profile_level && cond?.power_restriction_code) {
+        arrIndex = 1;
+      } else if (cond?.electrical_profile_level && !cond?.power_restriction_code) {
+        arrIndex = 2;
+      } else if (!cond?.electrical_profile_level && !cond?.power_restriction_code) {
+        arrIndex = 3;
+      }
+      sortedCurves[arrIndex].push(curve);
+      return sortedCurves;
+    },
+    [[], [], [], []] as ConditionalEffortCurve[][]
+  );
```

Comment:
Here you can use a object, I think it's more readable than an array (but the idea is the same). What do you think ?
```suggestion
  const { allConds, onlyPR, onlyEP, nulls } = curvesList.reduce(
    (sortedCurves, curve) => {
      const { cond } = curve;
      if (!cond) return sortedCurves;
      if (cond.electrical_profile_level && cond.power_restriction_code) {
        return { ...sortedCurves, allConds: [...sortedCurves.allConds, curve] };
      }
      if (cond.power_restriction_code) {
        return { ...sortedCurves, onlyPR: [...sortedCurves.allConds, curve] };
      }
      if (cond.electrical_profile_level) {
        return { ...sortedCurves, onlyEP: [...sortedCurves.allConds, curve] };
      }
      return { ...sortedCurves, nulls: [...sortedCurves.allConds, curve] };
    },
    { allConds: [], onlyPR: [], onlyEP: [], nulls: [] } as Record<string, ConditionalEffortCurve[]>
  );
```

Reply:
I agree :)